core/trino-main/src/main/java/io/trino/sql/gen/columnar/IsNullColumnarFilter.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.sql.gen.columnar;
+
+import com.google.common.collect.ImmutableList;
+import io.trino.operator.project.InputChannels;
+import io.trino.spi.Page;
+import io.trino.spi.block.Block;
+import io.trino.sql.relational.InputReferenceExpression;
+
+import java.util.List;
+
+public final class IsNullColumnarFilter
+        implements ColumnarFilter
+{
+    private final InputChannels inputChannels;
+
+    public IsNullColumnarFilter(InputReferenceExpression inputReferenceExpression)
+    {
+        List<Integer> channels = ImmutableList.of(inputReferenceExpression.getField());
+        this.inputChannels = new InputChannels(channels, channels);
+    }
+
+    @Override
+    public InputChannels getInputChannels()
+    {
+        return inputChannels;
+    }
+
+    @Override
+    public int filterPositionsRange(int[] outputPositions, int offset, int size, Page page)
+    {
+        Block block = page.getBlock(0);
```

Comment:
I would ensure that the caller unwraps dictionary and RLE by down casting to value block here.  

Also this and the isolation system are pretty complex just to get the nulls from a value block.  Instead we should just add a method to get the nulls as a BOOLEAN type block.  So add a method like `ValueBlock.getNulls()Optional<ValueBlock>`.  The method docs would require the returned value block many not have nulls, and is the same type a BooleanType... maybe it just returns ByteArrayBlock (or BooleanArrayBlock if we add that).  Once we have that, this code can grab the block, and then use the raw nulls array inside of it directly.  then we don't need all of the cermony in this system.  That said, we don't need to change that now, if you don't want to.

Reply:
I've added a `Optional<ByteArrayBlock> getNulls()` to `ValueBlock` and got rid of the classloader isolation logic